core/common.h

```diff
@@ -359,6 +359,8 @@ static inline void mbedtls_xor_no_simd(unsigned char *r,
  */
 #if defined(static_assert) && !defined(__FreeBSD__)
 #define MBEDTLS_STATIC_ASSERT(expr, msg)    static_assert(expr, msg)
+#elif defined(__GNUC__) && ((__GNUC__ == 4 && __GNUC_MINOR >= 6) || __GNUC__ > 4) && !defined(__cplusplus)
+#define MBEDTLS_STATIC_ASSERT(expr, msg)    _Static_assert(expr, msg)
```

Comment:
Sorry for the delay. After a few searches, it seems the following would support more compilers: 
```
#elif defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L && !defined(__cplusplus)
```

Reply:
This doesn't work for me, the issue is that gcc supports the static assertion even when you build with std=99 (from version >=4.6) Your version requires a newer std which is not always used in our code. Are you OK to add this statement in addition to the GNUC one? I think that this is the best case because then we cover both the gnu case and the std=11 case. I am happy to add that.

Reply:
`defined(static_assert)` just above should already cover C11.

Reply:
> This doesn't work for me, the issue is that gcc supports the static assertion even when you build with std=99 (from version >=4.6) Your version requires a newer std which is not always used in our code. Are you OK to add this statement in addition to the GNUC one? I think that this is the best case because then we cover both the gnu case and the std=11 case. I am happy to add that.

ah okay that makes sense. Yes I would say add both then with maybe your explanation here as a comment to explain why we need 
```#elif defined(__GNUC__) && ((__GNUC__ == 4 && __GNUC_MINOR >= 6) || __GNUC__ > 4) && !defined(__cplusplus)``` in addition to ```#elif defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L && !defined(__cplusplus)```.

Reply:
Ok cool, I checked for both and I added a comment explaining the check.